crates/utils/src/rate_limit/rate_limiter.rs

```diff
@@ -46,14 +46,16 @@ impl RateLimiter {
   }
 
   /// Rate limiting Algorithm described here: https://stackoverflow.com/a/668327/1655478
+  ///
+  /// Returns true if the request passed the rate limit, false if it failed and should be rejected.
   #[allow(clippy::float_cmp)]
   pub(super) fn check_rate_limit_full(
     &mut self,
     type_: RateLimitType,
     ip: &IpAddr,
     rate: i32,
     per: i32,
-  ) -> Result<(), LemmyError> {
+  ) -> Result<bool, LemmyError> {
```

Comment:
There's no `?`, so couldn't this whole function be changed from a result to just a bool?

Reply:
actually no, there is `current.duration_since(rate_limit.last_checked)?`

Reply:
okay i can get rid of that